### ISSUE TITLE
Repo overhaul

### ISSUE THREAD

**Issue by josh-chamberlain** [Police-Data-Accessibility-Project/PDAP-Scrapers#197]
# Scrapers repo

> This stores our common scrapers which don’t have their own repo or a better home.
> These serve as examples of different ways to get at data.
> Find other scrapers by their metadata

## Problems with the current repo
- too intimidating
- too complex
- people aren't sure how to contribute
   - people don't know what the code is for
   - there's no good example of how the code can be used

## How people use the repo
- find a way to help
- run scrapers locally
    - find them geographically
    - sorted by language
- people can run utilities for writing scrapers
    - common files
- serve as a library of other scrapers

## To do

<details>
<summary>Readme changes</summary>

- [ ] welcome to PDAP! [it should feel like this](https://github.com/ohmyzsh/ohmyzsh)
- [ ] add a visual representation of how this relates to our other work
   - scrapers outside this repo
   - data sources
   - work being done with data
- [ ] add a visual representation of this repo
   - how do scrapers relate to common files
   - what is contained in each scraper's folder
- [ ] a high level explanation of what to expect when code in the repo is run
   - [ ] examples of things that work
       - show the depth + breadth of our work
          - different strategies for scraping
          - different record types + ways to process them
          - different geographies
   - [ ] what are utilities, and why should you use them?
   - [ ] what are scrapers, and why should you use them?
</details>

<details>
<summary>Structure changes</summary>

- [ ] support languages other than Python
   - we could organize all scrapers by language first in parent folders for each language
   - we could make companion repos without the redundant stuff for any other languages which come up; this main one would be for python, which is the most common language.
- [ ] merge/combine directories so that it's more clear what things do and how they relate
   - update the readme as part of this

</details>

## Related work
https://github.com/Police-Data-Accessibility-Project/PDAP-Scrapers/issues/196

**Comment by EvilDrPurple** [comment]
Just want a clarification real quick, you put:

```
scrapers/
    -- data_portals/
    -- cityprotect/
    -- crimegraphics/
         -- README.md
         -- crimegraphics.py
  -- federal/
```
You indented this more than the other lines so I'm assuming you want what is currently the top-level [Base_Scripts ](https://github.com/Police-Data-Accessibility-Project/PDAP-Scrapers/tree/main/Base_Scripts) directory to go inside the scrapers folder, or did you want them inside the `utils` folder instead?


> one major reason for this is that the "common" scripts are all over the place, so for a new user it's incredibly difficult to figure out how they relate.

Based on this line I'm also assuming you want some of the duplicate scripts in various places removed and streamlined to one area. Just want to make sure this is what you meant

**Comment by josh-chamberlain** [comment]
@EvilDrPurple yes, the duplicate scripts should each have one home. I grouped some of them under `data_portals/` and some under `utils/` because there are both kinds, and they do different things—data portal scrapers can be run on any portal built on that platform, the utils do other random stuff.

I know it might be a pain taking something that's currently spread across 3 random directories and refactoring it to work in just 1. As you work, if you find things that are broken when you get there, let's talk about whether it's worth fixing for the sake of reorganizing the repo vs. just pulling it out / putting it on ice somehow.
